App/xcbuddykit/Sources/Loader/GraphModuleLoader.swift

```diff
@@ -0,0 +1,91 @@
+import Basic
+import Foundation
+
+enum GraphModuleLoaderError: FatalError {
+    case fileNotFound(AbsolutePath)
+    case fileLoadError(AbsolutePath, Error)
+
+    /// Error type.
+    var type: ErrorType {
+        switch self {
+        case .fileNotFound:
+            return .abort
+        case .fileLoadError:
+            return .bugSilent
+        }
+    }
+
+    /// Error description.
+    var description: String {
+        switch self {
+        case let .fileNotFound(path):
+            return "Couldn't file file at path \(path)"
+        case let .fileLoadError(path):
+            return "Error loading file at path \(path)"
+        }
+    }
+}
+
+/// Interface of a module loader.
+protocol GraphModuleLoading: AnyObject {
+    /// Given a path, it loads all the paths that should be part of the module.
+    /// Any file can include other files in the same module by adding a comment line at
+    /// the top:
+    ///    //include: ./shared.swift
+    /// Notice that the path needs to be relative to the file.
+    ///
+    /// - Parameter path: path whose module will be loaded.
+    /// - Parameter context: context.
+    /// - Returns: list of files that should be part of the module.
+    func load(_ path: AbsolutePath, context: Contexting) throws -> Set<AbsolutePath>
+}
+
+final class GraphModuleLoader: GraphModuleLoading {
+    /// Regular expression used to identify include lines.
+    // swiftlint:disable:next force_try
+    static let regex = try! NSRegularExpression(pattern: "//\\s*include:\\s+\"?(.+)\"?", options: [])
+
+    /// Given a path, it loads all the paths that should be part of the module.
+    /// Any file can include other files in the same module by adding a comment line at
+    /// the top:
+    ///    //include: ./shared.swift
+    /// Notice that the path needs to be relative to the file.
+    ///
+    /// - Parameter path: path whose module will be loaded.
+    /// - Parameter context: context.
+    /// - Returns: list of files that should be part of the module.
+    func load(_ path: AbsolutePath, context: Contexting) throws -> Set<AbsolutePath> {
+        var paths = Set<AbsolutePath>()
+        try load(path, paths: &paths, context: context)
+        return paths
+    }
+
+    /// Recursive method that traverses the module files and adds its paths to the given paths set.
+    ///
+    /// - Parameters:
+    ///   - path: path to be inspected.
+    ///   - paths: set that contains the module paths.
+    ///   - context: context.
+    fileprivate func load(_ path: AbsolutePath, paths: inout Set<AbsolutePath>, context: Contexting) throws {
+        var paths = Set<AbsolutePath>()
```

Comment:
Is this the final implementation of this method? I have some doubts:
- Why you don't use the `paths` argument?
- When are you adding new element to the set?

Reply:
- You are right, we don't need to create any instance here.
- Lol, I forgot about it.

Reply:
That's why it's good to write tests. I would have caught it before the review. I'm working on adding them.